rust/core-lib/src/view.rs

```diff
@@ -73,20 +69,11 @@ pub struct View {
     /// description for the invariant.
     lc_shadow: LineCacheShadow,
 
-    /// The occurrences, which determine the highlights, have been updated.
-    hls_dirty: bool,
-
     /// New offset to be scrolled into position after an edit.
     scroll_to: Option<usize>,
 
-    /// The currently active search string
-    search_string: Option<String>,
-    /// The case matching setting for the currently active search
-    case_matching: CaseMatching,
-    /// The set of all known find occurrences (highlights)
-    occurrences: Option<Selection>,
-    /// Set of ranges that have already been searched for the currently active search string
-    valid_search: IndexSet,
+    /// The state for finding text for this view.
+    find: Find,
```

Comment:
Semantically, I wonder if this shouldn't be an `Option<Find>`?

Reply:
If we have one `Find` instance for each query then they would be stored in `Vec` otherwise `Option<Find>`.